### PR TITLE
[40946173] Allow custom email layouts per project and custom email templates per protocol

### PR DESCRIPTION
**Pull request by emerencia**
Fixes https://freedcamp.com/view/2914531/tasks/panel/task/40946173

### CODE REVIEW

File: app/mailers/invitation_mailer.rb [hunk 1]
```diff
@@ -8,17 +8,23 @@ class InvitationMailer < ApplicationMailer
...
     @message = message
-    mail(subject: DEFAULT_INVITATION_SUBJECT, to: email_address)
+    # Use the default template if the given template does not exist.
+    template = lookup_context.find_all(template, []).blank? ? 'invitation_mailer/invitation_mail' : template
+    mail(subject: DEFAULT_INVITATION_SUBJECT, to: email_address) do |format|
+      format.html { render template: template }
+    end
   end
 
+  # Only used for u-can-act symposium registration. Remove when removing the u-can-act project.
```
Comment (m-rios) [inline app/mailers/invitation_mailer.rb:21]: maybe prepend a TODO so it's easier to find in the future?

File: app/mailers/invitation_mailer.rb [hunk 2]
```diff
@@ -8,17 +8,23 @@ class InvitationMailer < ApplicationMailer
...
   def confirmation_mail(email_address, subject, message)
     @message = message
     mail(subject: subject, to: email_address)
   end
 
+  # Only used for the IKIA project for parents to invite children. Remove when removing the ikia project.
```
Comment (m-rios) [inline app/mailers/invitation_mailer.rb:27]: same, TODO

**Review by m-rios — APPROVED** [review]

File: app/mailers/invitation_mailer.rb (hunk 1 above)
Comment (emerencia) [inline app/mailers/invitation_mailer.rb:21]:
```suggestion
  # TODO: Only used for u-can-act symposium registration. Remove when removing the u-can-act project.
```

File: app/mailers/invitation_mailer.rb (hunk 2 above)
Comment (emerencia) [inline app/mailers/invitation_mailer.rb:27]:
```suggestion
  # TODO: Only used for the IKIA project for parents to invite children. Remove when removing the ikia project.
```

Comment (emerencia) [reply app/mailers/invitation_mailer.rb]: Fixed.

File: app/mailers/invitation_mailer.rb (hunk 1 above)
Comment (emerencia) [reply app/mailers/invitation_mailer.rb]: Fixed.

**Pull request merged**